src/components/compCity/weatherHeader.js

```diff
@@ -1,8 +1,32 @@
 import TextWeatherContent from './textWeatherContent';
+import { useSession, signIn } from 'next-auth/react';
+import { useState } from 'react';
 import { HStack, VStack, Button, Center, Text } from '@chakra-ui/react';
 import { LiaStarSolid } from 'react-icons/lia';
 
-export default function WeatherHeader({ weatherData }) {
+export default function WeatherHeader({
+  weatherData,
+  addToFavorites,
+  deleteCityFromFavorites,
+  cityIsSavedToFavorites,
+}) {
+  const [isToggled, setIsToggled] = useState(false);
+  const { data: session } = useSession();
+
+  function handleClick() {
+    if (!session) {
+      signIn();
+    }
+    if (session && !cityIsSavedToFavorites) {
+      addToFavorites();
```

Comment:
I think you can define addToFavorites and deleteCityFromFavorites functions in this component, and not in his parent. Also, you can remove cityIsSavedToFavorites from the parent component, because you already have isToggled here.